### ISSUE TITLE
Python-magic potentially conflicts with file-magic

### ISSUE THREAD

**Issue by blubard** [CZ-NIC/envelope#25]
Using the envelope library in a bigger project might lead to namespace clashes due to envelope's use of python-magic. If a project somehow has a dependency on file-magic (which has identical or similar functionality but is used differently) there is not guarantee for the whole project to work, as both are imported via `import magic`.

Including either of the libraries via extra requirements could solve this potential issue.

**Comment by e3rd** [comment]
Thanks for pointing that out! I'll take a look at it but not earlier than
on next Wed.

**Comment by e3rd** [comment]
Hi, I dive a little bit inside, it's a pity they use the same keyword. By extra requirements you mean `extras_require` in the `setup.py`? Like do not use the funcionality if not installed? Or how would you implement it?

**Comment by blubard** [comment]
Yes, I meant the `extras_require`. Preferably, you would use a default (likely the current version, in order to avoid problems for your current users) and allow a version that enforces python-magic via envelope[python-magic]? (not sure if via extras you can avoid the default dependency though?)

**Comment by blubard** [comment]
Apparently you can't have a default and replace it via extras_require. It appears that enforcing an extra requirement is the cleanest solution.

**Comment by e3rd** [comment]
I'm a bit unhappy about this issue, I see no clean solution. It seems to me if I put `python-magic` inside `extras_require`, all that it means is that it is not guaranteed the envelope will work when launched. Because `import magic` could still work while having another API. :(
